File: rust/dvec1.rs
fn f(n: uint) -> uint {
  let mut c : uint = 0;
  let v : dvec::DVec<uint> = dvec::DVec();

  while c < n {
    v.push(c);
    c += 1;
  }

  assert v.len() == n;
  return v.get_elt(rand::random() % v.len());
}

fn main() {
  let args = os::args();
  let n = uint::from_str(args[1]).get();
  let r = uint::from_str(args[2]).get();

  let v : dvec::DVec<uint> = dvec::DVec();
  let mut c : uint = 0;
  while c < r {
    v.push(f(n));
    c += 1;
  }

  io::println(fmt!("%?", v.get_elt(0)));
}

File: go/dvec1.go
package main

import "fmt"
import "os"
import "strconv"
import "math/rand"

func f(n uint) uint {
  v := make([]uint, 0)
  var c uint = 0
  for ; c < n; c++ {
    v = append(v, c) 
  }
  return v[rand.Intn(int(n))]
}

func main() {
  n, err := strconv.ParseUint(os.Args[1], 10, 64)
  if err != nil {
    return
  }

  r, err := strconv.ParseUint(os.Args[2], 10, 64)
  if err != nil {
    return
  }

  v := make([]uint, 0)
  var c uint64 = 0
  for ; c < r; c++ {
    v = append(v, f(uint(n)))
  }

  fmt.Println(v[0]) 
}
